### ISSUE TITLE
Regression in `ReverseDiff.increment_deriv!`

### ISSUE THREAD

**Issue by ztangent** [probcomp/Gen.jl#415]
[Recent Travis builds](https://github.com/probcomp/Gen.jl/runs/2576319358) have started failing because of an issue with `ReverseDiff.increment_deriv!`:

```julia
mixture of multivariate normals: Error During Test at /home/travis/build/probcomp/Gen.jl/test/modeling_library/mixture.jl:138
587  Got exception outside of a @test
588  MethodError: no method matching increment_deriv!(::Float64, ::Float64)
589  Closest candidates are:
590    increment_deriv!(!Matched::ReverseDiff.TrackedArray, ::Real, !Matched::Any) at /home/travis/.julia/packages/ReverseDiff/60noS/src/derivatives/propagation.jl:34
591    increment_deriv!(!Matched::AbstractArray, ::Real, !Matched::Any) at /home/travis/.julia/packages/ReverseDiff/60noS/src/derivatives/propagation.jl:36
592    increment_deriv!(!Matched::AbstractArray, ::Any) at /home/travis/.julia/packages/ReverseDiff/60noS/src/derivatives/propagation.jl:39
  Stacktrace:
   [1] increment_deriv! at /home/travis/.julia/packages/ReverseDiff/60noS/src/derivatives/propagation.jl:35 [inlined]
   [2] increment_deriv!(::Array{Real,3}, ::Array{Float64,3}) at /home/travis/.julia/packages/ReverseDiff/60noS/src/derivatives/propagation.jl:40
   [3] increment_deriv!(::Array{Real,3}, ::Array{Float64,3}) at /home/travis/build/probcomp/Gen.jl/src/backprop.jl:16
   [4] special_reverse_exec!(::ReverseDiff.SpecialInstruction{HomogeneousMixture{Array{Float64,1}},Tuple{Array{Float64,1},Array{Float64,1},Array{Float64,2},Array{Real,3}},ReverseDiff.TrackedReal{Float64,Float64,Nothing},Nothing}) at /home/travis/build/probcomp/Gen.jl/src/dynamic/backprop.jl:24
```

This appears to be an upstream issue introduced in ReverseDiff 1.8.0:
https://github.com/JuliaDiff/ReverseDiff.jl/issues/175

It should be fixed once the following PR is merged:
https://github.com/JuliaDiff/ReverseDiff.jl/pull/177

In the meantime, users running into this bug should downgrade to ReverseDiff 1.7.0.

**Comment by bzinberg** [comment]
In theory, we could work around this by adding `ReverseDiff = "<= 1.7.0"` to the `[compat]` section of Gen's `Project.toml`.  In this case not worth it, since looks like that fix is about to be merged upstream.  So, once it is, we should add `ReverseDiff = "<the release that includes that PR>"` :slightly_smiling_face:

**Comment by ztangent** [comment]
Addressed by JuliaDiff/ReverseDiff.jl#177 being merged, and ReverseDiff 1.9.0 being released. Will open a separate issue for restricting version compatibility to prevent issues like these going forward.
